src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -395,21 +397,45 @@ private static async Task<string> DownloadBuildArtifactsAsync(BuildHttpClient bu
             var archiveDownloadPath = Path.Combine(tempDirectory, artifact.Name);
             Console.WriteLine($"Downloading artifacts to {archiveDownloadPath}");
 
-            Stopwatch watch = Stopwatch.StartNew();
+            var artifactType = artifact.Resource.Type;
+            Debug.Assert(artifactType.Equals("container", StringComparison.OrdinalIgnoreCase) || artifactType.Equals("pipelineArtifact", StringComparison.OrdinalIgnoreCase));
 
-            using (Stream s = await buildClient.GetArtifactContentZipAsync(Options.ComponentBuildProjectNameOrFallback, build.Id, artifact.Name, cancellationToken))
-            using (var ms = new MemoryStream())
+            Stopwatch watch = Stopwatch.StartNew();
+            if (artifactType.Equals("container", StringComparison.OrdinalIgnoreCase))
             {
-                await s.CopyToAsync(ms);
-                using (ZipArchive archive = new ZipArchive(ms))
-                {
-                    archive.ExtractToDirectory(tempDirectory);
-                }
+                using var contentStream = await buildClient.GetArtifactContentZipAsync(Options.ComponentBuildProjectNameOrFallback, build.Id, artifact.Name, cancellationToken);
```

Comment:
This is just the original logic.

Reply:
It looks like copy to MemoryStream was removed, was this just felt to be an unnecessary step? (and I guess one that might consume a lot of memory..)